crates/core/src/factories/core_fetch.rs

```diff
@@ -166,20 +167,32 @@ impl Fetch for CoreFetch {
         op_ids: Vec<OpId>,
         source: Url,
     ) -> BoxFut<'_, K2Result<()>> {
-        // Add requests to set.
-        {
-            let requests = &mut self.state.lock().unwrap().requests;
-            requests.extend(
-                op_ids
-                    .clone()
-                    .into_iter()
-                    .map(|op_id| (op_id.clone(), source.clone())),
-            );
-        }
-
         Box::pin(async move {
+            // Filter out requests for ops that are already in the op store.
+            let existing_op_ids = self
+                .op_store
+                .retrieve_ops(op_ids.clone())
```

Comment:
This requires loading the ops out of the database and unpacking the msgpack blobs. Should we add an operation to the op store API that just checks if those ops are present?

Reply:
I like that. I was hesitant to add another operation. But there's no good reason.

Reply:
How about now?